### PR TITLE
fix: account for packument with no versions

### PR DESCRIPTION
**Pull request by yadirhb**
Details: https://github.com/npm/metavuln-calculator/issues/80

### CODE REVIEW

**Comment by wraithgar** [comment]
Linting and tests need to pass.

File: lib/advisory.js [hunk 1]
```diff
@@ -242,7 +242,7 @@ class Advisory {
     // check the dependency of this version on the vulnerable dep
     // if we got a version that's not in the packument, fall back on
     // the spec provided, if possible.
-    const mani = this[_packument].versions[version] || {
+    const mani = this[_packument].versions && this[_packument].versions[version] ? this[_packument].versions[version] : {
```
Comment (vigan-abd) [inline lib/advisory.js:245]: hi @yadirhb, thanks a lot for contributing to this issue, if possible could you try to fix this linting issue so git flow will pass?

Comment (rainabba) [reply lib/advisory.js:245]: It just needs line-breaks:

```
const mani = this[_packument].versions && this[_packument].versions[version] ? 
    this[_packument].versions[version] : 
    {
```

I submitted [this PR](https://github.com/yadirhb/metavuln-calculator/pull/1) a few days ago to address that. Should I setup a new one here to get this moving?

Comment (vigan-abd) [reply lib/advisory.js:245]: https://github.com/yadirhb/metavuln-calculator/pull/1/files#r1147585254

Comment (vigan-abd) [reply lib/advisory.js:245]: tbh I'm not sure if original author is available to accept your changes or if opening a new PR would help :/

**Comment by wraithgar** [comment]
Packuments only have no versions when they've been unpublished, and wouldn't be in your tree to be send to this module in the first place.

**Comment by rainabba** [comment]
"wouldn't be in your tree "

@wraithgar I hear your theory, but do you think it's more likely that more than one of us landed here at the same time, for the same reason, to that very line of code; if it weren't running and causing the error we've reported? It may be that it "shouldn't" happen, but if it does, do you want to continue to be responsible for maintaining fragile code, or do you want to AT LEAST let the logical failure move on to be exposed higher up the call-chain? Right now, this code IS the problem and to me, you're coming across very short-termed and extremely unapreciative of the time we're putting in to try and help.

I'm open to the possibility that I'm still confused, but it seems a lot less likely to me. Perhaps you're assuming (as the code does), that a scenario will not occur that IS occurring and because the code isn't as robust as it could be, we're here today?

**Comment by vigan-abd** [comment]
hey @wraithgar true but this happens also when you try to install a package via github link, and previously some package with same name existed in npm (e.g. a fake package), and this is causing issues. Due to this I would suggest to add a support for this optional versions field.

Sample case is this package: https://github.com/bitfinexcom/bfx-svc-test-helper, someone put a fake package with same name previously on npm and it got unpublished as expected, but now due to this npm issue we can't install this dependency from github

**Comment by rainabba** [comment]
So is this bug going to be left in the codebase or...? We have multieple pipelines failing on `npm i` still becaues of this.

**Comment by wraithgar** [comment]
The underlying bug is https://github.com/npm/cli/issues/5110, it is not a bug with this module.

**Pull request closed without merging**